doc/arch/adr-001-notify-integration.md

```diff
@@ -0,0 +1,424 @@
+   relate
+2. [Sequence diagram](../sequence_diagram.png) which represents all of the
+   components and how they communicate to achieve specified functionality
+
+This document then goes through each of the concepts in the diagram and outline
+their role and responsibility.
+
+### ContentChange
+
+**Responsibility**: Representing a piece of content that is either new or has
+changed and should produce an alert.
+
+#### Shall
+
+- Store information such as:
+  - Title of content
+  - Description of content
+  - Change note (if any) associated with the update
+  - Path to content
+
+#### Should
+
+- Store information about what subscription criteria it matches (links, tags,
+  supertypes)
+- Have fields required for auditing
+  - Where the request came from
+  - Be able to determine whether and when it has been processed by a worker
+    (processed_at presence)
+- Be persisted forever
+
+#### May
+
+- In future, have an association to an entity that models the subscription
+  criteria information (links, tags, super types)
+- Be considered as part of an Alert supertype which may represent other types
+  of alerts in the system
+
+### Subscribable
+
+**Responsibility**: Being a type of list that a user can subscribe to and storing
+information specific to that list.
+
+#### Shall
+
+- Store information such as
+  - List name
+
+#### Should
+
+- Require that the list name is unique
+- Maintain a relationship with corresponding govdelivery list
+- Store information about what subscription criteria it matches
+  (links, tags, document_supertypes)
+- Have information to source where the Subscribable originated from
+
+#### May
+
+- In future, have an association to an entity that models the subscription
+  criteria information (links, tags, super types)
+- Release the requirement of a list name being unique if we are able to
+  establish a way to differentiate between lists
+- Store statistics related to the list
+- Be stored even after it is removed - for logging purposes, dependent on if
+  this will impact unique constraints
+
+#### To determine
+
+- In what scenarios is a Subscribable removed
+
+### Subscriber
+
+**Responsibility**: Represent an individual user that can subscribe to content,
+stores information related to all subscriptions the user may have
+
+#### Shall
+
+- Store information such as
+  - Email Address
+- Support an email address changing
+
+#### Should
+
+- Be deleteable, a user may wish to remove their account - this should not
+  break history of emails sent
+- Be used in the future for authentication purposes
+
+#### May
+
+- Store preferences that affect all of a users subscriptions
+- Have the ability to be disabled if there are problems with a users account
+- Store other personalisation information such as name
+- Be associated with statistics regarding the subscriber
+
+#### To determine
+- What should we allow in terms of removing a subscriber - do we use the
```

Comment:
An IA question here: do we need to completely purge a user if they delete themselves from the system? (Guessing yes since we no longer have a reason to store their details). I think this will come out of the IA discussions we're having at the moment.

Reply:
Also interesting is what we do with users who have subscribed to some emails and then unsubscribed from everything. Do we auto-remove? Also something to think about in the future if/when we are relying on links from emails essentially being the user's way to access their account; a user unsubscribed from everything no longer gets any emails and may not be aware that their data is still stored on our system....or may be fairly confused about how to get back to delete their account.

Reply:
Yes these are interesting problems.

@rubenarakelyan yup be great if we can worth that out. Is this something already on an agenda for discussions or should we log it somewhere?

@bevanloon I assume that the account still exists even if there are no subscriptions, it helps maintain our logs better and can offer benefits of maintaining users preferences. I don't know though if this factors into IA if we have any responsibility to remove data after a while.

Reply:
@kevindew Having spoken to Tim, we should just blank out the email address of the user if they unsubscribe from everything. We can still keep the record since other things rely on it but then it no longer has any personal information in it. I've created a card to do this for planning.